handle build warnings

The VS/Windows build shows a number of warnings:
- warning C4316, 20
- warning C4018, 25
- warning LNK4221, 4

All warnings need to be addressed.
Double check Linux and macOS.

@syncle there are 27 type conversion warnings in ColorMapOptimization.cpp. You can get this list of warnings by compiling with Visual Studio 2015.

Can you write explicit type conversions and make sure the logic is correct? (I.e., none of the type conversion is not as algorithm designed.)

```
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(104): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(105): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
8>  jquant2.c
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(106): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
8>  jutils.c
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(132): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
8>  Generating Code...
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(344): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(345): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(349): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(350): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(351): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(384): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(385): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(386): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(387): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(388): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(389): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(390): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(391): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
16>  pngread.c
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(488): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(489): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(493): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(496): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(497): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
16>  pngrio.c
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(205): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
9>  D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(236): note: see reference to function template instantiation 'std::tuple<bool,float> three::`anonymous-namespace'::QueryImageIntensity<float>(const three::Image &,const three::`anonymous-namespace'::ImageWarpingField &,const Eigen::Vector3d &,const three::PinholeCameraTrajectory &,int,int)' being compiled
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(208): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(182): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
9>  D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(267): note: see reference to function template instantiation 'std::tuple<bool,float> three::`anonymous-namespace'::QueryImageIntensity<float>(const three::Image &,const Eigen::Vector3d &,const three::PinholeCameraTrajectory &,int,int)' being compiled
9>D:\Research\Open3D\src\Core\ColorMap\ColorMapOptimization.cpp(184): warning C4244: 'argument': conversion from 'float' to 'int', possible loss of data
```

--- a/docs/tutorial/Advanced/color_map_optimization.rst
+++ b/docs/tutorial/Advanced/color_map_optimization.rst
@@ -55,7 +55,7 @@ Consider color mapping to the geometry reconstructed from depth cameras. As colo
             # Q.-Y. Zhou and V. Koltun,
             # Color Map Optimization for 3D Reconstruction with Consumer Depth Cameras,
             # SIGGRAPH 2014
-            option.maximum_iteration = 500
+            option.maximum_iteration = 300
             option.non_rigid_camera_coordinate = True
             color_map_optimization(mesh, rgbd_images, camera, option)
             draw_geometries([mesh])
@@ -121,25 +121,25 @@ The next step is to optimize camera poses to get a sharp color map.
 
 .. code-block:: python
 
-    option.maximum_iteration = 500
+    option.maximum_iteration = 300
     option.non_rigid_camera_coordinate = True
     color_map_optimization(mesh, rgbd_images, camera, option)
     draw_geometries([mesh])
     write_triangle_mesh(os.path.join(path, "scene",
         "color_map_after_optimization.ply"), mesh)
 
-The script sets ``maximum_iteration = 500`` for actual iterations. The optimization displays the following energy profile.
+The script sets ``maximum_iteration = 300`` for actual iterations. The optimization displays the following energy profile.
 
 .. code-block:: shell
 
     [ColorMapOptimization] :: Rigid Optimization
-    [Iteration 0001] Residual error : 25777.372725 (avg : 0.004998)
-    [Iteration 0002] Residual error : 25620.681829 (avg : 0.004967)
-    [Iteration 0003] Residual error : 25463.806101 (avg : 0.004937)
+    [Iteration 0001] Residual error : 21639.276499 (avg : 0.004615)
+    [Iteration 0002] Residual error : 21461.765357 (avg : 0.004577)
+    [Iteration 0003] Residual error : 21284.579715 (avg : 0.004540)
     :
-    [Iteration 0498] Residual error : 11550.014763 (avg : 0.002255)
-    [Iteration 0499] Residual error : 11549.850827 (avg : 0.002255)
-    [Iteration 0500] Residual error : 11550.062068 (avg : 0.002255)
+    [Iteration 0298] Residual error : 8891.042884 (avg : 0.001903)
+    [Iteration 0299] Residual error : 8890.037077 (avg : 0.001903)
+    [Iteration 0300] Residual error : 8888.970765 (avg : 0.001903)
 
 Residual error implies inconsistency of image intensities. Lower residual leads better color map quality. By default, ``ColorMapOptmizationOption`` enables rigid optimization. It optimizes 6-dimentional pose of every cameras.
 
@@ -163,13 +163,13 @@ before calling ``color_map_optimization``. Besides 6-dimentional camera poses, n
 .. code-block:: shell
 
     [ColorMapOptimization] :: Non-Rigid Optimization
-    [Iteration 0001] Residual error : 25777.372725, reg : 0.000000
-    [Iteration 0002] Residual error : 25330.445704, reg : 13.005639
-    [Iteration 0003] Residual error : 24885.912182, reg : 40.000765
+    [Iteration 0001] Residual error : 21639.276499, reg : 0.000000
+    [Iteration 0002] Residual error : 21187.225206, reg : 13.918495
+    [Iteration 0003] Residual error : 20745.248996, reg : 42.234724
     :
-    [Iteration 0498] Residual error : 7585.606850, reg : 3294.124184
-    [Iteration 0499] Residual error : 7585.274846, reg : 3294.887659
-    [Iteration 0500] Residual error : 7583.972930, reg : 3294.634065
+    [Iteration 0298] Residual error : 5589.018747, reg : 2745.364742
+    [Iteration 0299] Residual error : 5587.180145, reg : 2746.619137
+    [Iteration 0300] Residual error : 5585.066255, reg : 2747.902979
 
 Results of non-rigid optimization follow.
 

--- a/src/Core/ColorMap/ColorMapOptimization.cpp
+++ b/src/Core/ColorMap/ColorMapOptimization.cpp
@@ -124,7 +124,7 @@ std::tuple<std::vector<std::vector<int>>, std::vector<std::vector<int>>>
         int viscnt = 0;
         for (int vertex_id = 0; vertex_id < n_vertex; vertex_id++) {
             Eigen::Vector3d X = mesh.vertices_[vertex_id];
-            double u, v, d;
+            float u, v, d;
             std::tie(u, v, d) = Project3DPointAndGetUVDepth(X, camera, c);
             int u_d = int(round(u)), v_d = int(round(v));
             if (d < 0.0 || !images_rgbd[c].depth_.TestImageBoundary(u_d, v_d))
@@ -134,7 +134,7 @@ std::tuple<std::vector<std::vector<int>>, std::vector<std::vector<int>>>
                 continue;
             if (*PointerAt<unsigned char>(images_mask[c], u_d, v_d) == 255)
                 continue;
-            if (abs(d - d_sensor) <
+            if (std::fabs(d - d_sensor) <
                     option.depth_threshold_for_visiblity_check_) {
 #ifdef _OPENMP
 #pragma omp critical

--- a/src/Core/ColorMap/ColorMapOptimization.cpp
+++ b/src/Core/ColorMap/ColorMapOptimization.cpp
@@ -36,9 +36,6 @@
 #include <IO/ClassIO/TriangleMeshIO.h>
 #include <Core/Utility/Eigen.h>
 
-// for debugging
-#include <IO/ClassIO/ImageIO.h>
-
 namespace three {
 
 namespace {
@@ -53,8 +50,8 @@ public:
     void InitializeWarpingFields(int width, int height,
             int number_of_vertical_anchors) {
         anchor_h_ = number_of_vertical_anchors;
-        anchor_step_ = (double)height / (anchor_h_ - 1);
-        anchor_w_ = std::ceil((double)width / anchor_step_) + 1;
+        anchor_step_ = double(height) / (anchor_h_ - 1);
+        anchor_w_ = int(std::ceil(double(width) / anchor_step_) + 1);
         flow_ = Eigen::VectorXd(anchor_w_ * anchor_h_ * 2);
         for (int i = 0; i <= (anchor_w_ - 1); i++) {
             for (int j = 0; j <= (anchor_h_ - 1); j++) {
@@ -101,9 +98,9 @@ inline std::tuple<float, float, float> Project3DPointAndGetUVDepth(
     std::pair<double, double> p = camera.intrinsic_.GetPrincipalPoint();
     Eigen::Vector4d Vt = camera.extrinsic_[camid] *
             Eigen::Vector4d(X(0), X(1), X(2), 1);
-    float u = (Vt(0) * f.first) / Vt(2) + p.first;
-    float v = (Vt(1) * f.second) / Vt(2) + p.second;
-    float z = Vt(2);
+    float u = float((Vt(0) * f.first) / Vt(2) + p.first);
+    float v = float((Vt(1) * f.second) / Vt(2) + p.second);
+    float z = float(Vt(2));
     return std::make_tuple(u, v, z);
 }
 
@@ -129,7 +126,7 @@ std::tuple<std::vector<std::vector<int>>, std::vector<std::vector<int>>>
             Eigen::Vector3d X = mesh.vertices_[vertex_id];
             double u, v, d;
             std::tie(u, v, d) = Project3DPointAndGetUVDepth(X, camera, c);
-            int u_d = round(u), v_d = round(v);
+            int u_d = int(round(u)), v_d = int(round(v));
             if (d < 0.0 || !images_rgbd[c].depth_.TestImageBoundary(u_d, v_d))
                 continue;
             float d_sensor = *PointerAt<float>(images_rgbd[c].depth_, u_d, v_d);
@@ -178,10 +175,14 @@ std::tuple<bool, T> QueryImageIntensity(
     float u, v, depth;
     std::tie(u, v, depth) = Project3DPointAndGetUVDepth(V, camera, camid);
     if (img.TestImageBoundary(u, v, IMAGE_BOUNDARY_MARGIN)) {
+		int u_round = int(round(u));
+		int v_round = int(round(v));
         if (ch == -1) {
-            return std::make_tuple(true, *PointerAt<T>(img, u, v));
+            return std::make_tuple(true, 
+					*PointerAt<T>(img, u_round, v_round));
         } else {
-            return std::make_tuple(true, *PointerAt<T>(img, u, v, ch));
+            return std::make_tuple(true, 
+					*PointerAt<T>(img, u_round, v_round, ch));
         }
     } else {
         return std::make_tuple(false, 0);
@@ -200,12 +201,14 @@ std::tuple<bool, T> QueryImageIntensity(
         Eigen::Vector2d uv_shift = field.GetImageWarpingField(u, v);
         if (img.TestImageBoundary(uv_shift(0), uv_shift(1),
                 IMAGE_BOUNDARY_MARGIN)) {
+			int u_shift = int(uv_shift(0));
+			int v_shift = int(uv_shift(1));
             if (ch == -1) {
                 return std::make_tuple(true,
-                        *PointerAt<T>(img, uv_shift(0), uv_shift(1)));
+                        *PointerAt<T>(img, u_shift, v_shift));
             } else {
                 return std::make_tuple(true,
-                        *PointerAt<T>(img, uv_shift(0), uv_shift(1), ch));
+                        *PointerAt<T>(img, u_shift, v_shift, ch));
             }
         }
     }
@@ -318,7 +321,7 @@ void OptimizeImageCoorNonrigid(
             Eigen::Matrix4d pose;
             pose = camera.extrinsic_[i];
             double anchor_step = warping_fields[i].anchor_step_;
-            double anchor_w = warping_fields[i].anchor_w_;
+            int anchor_w = warping_fields[i].anchor_w_;
             for (auto iter = 0; iter < visiblity_image_to_vertex[i].size();
                     iter++) {
                 int j = visiblity_image_to_vertex[i][iter];
@@ -341,14 +344,16 @@ void OptimizeImageCoorNonrigid(
                     + (1 - p) * (q)* grids[1]
                     + (p)* (1 - q) * grids[2]
                     + (p)* (q)* grids[3];
-                float uu = uuvv(0);
-                float vv = uuvv(1);
+                double uu = uuvv(0);
+                double vv = uuvv(1);
                 if (!images_gray[i]->TestImageBoundary(uu, vv,
                         IMAGE_BOUNDARY_MARGIN))
                     continue;
-                double gray = *PointerAt<float>(*images_gray[i], uu, vv);
-                Eigen::Vector2d dIdf(*PointerAt<float>(*images_dx[i], uu, vv),
-                        *PointerAt<float>(*images_dy[i], uu, vv));
+				bool valid; double gray, dIdfx, dIdfy;
+				std::tie(valid, gray) = images_gray[i]->FloatValueAt(uu, vv); 
+				std::tie(valid, dIdfx) = images_dx[i]->FloatValueAt(uu, vv);
+				std::tie(valid, dIdfy) = images_dy[i]->FloatValueAt(uu, vv);
+                Eigen::Vector2d dIdf(dIdfx, dIdfy);
                 Eigen::Vector2d dfdx = ((grids[2] - grids[0]) * (1 - q) +
                         (grids[3] - grids[1]) * q) / anchor_step;
                 Eigen::Vector2d dfdy = ((grids[1] - grids[0]) * (1 - p) +
@@ -485,16 +490,17 @@ void OptimizeImageCoorRigid(
                 Eigen::Vector3d V = mesh.vertices_[j];
                 Eigen::Vector4d G = pose * Eigen::Vector4d(V(0), V(1), V(2), 1);
                 Eigen::Vector4d L = intr * G;
-                float u = L(0) / L(2);
-                float v = L(1) / L(2);
+                double u = L(0) / L(2);
+                double v = L(1) / L(2);
                 if (!images_gray[i]->TestImageBoundary(u, v,
                         IMAGE_BOUNDARY_MARGIN))
                     continue;
-                double gray = *PointerAt<float>(*images_gray[i], u, v);
+				bool valid; double gray, dIdx, dIdy;
+				std::tie(valid, gray) = images_gray[i]->FloatValueAt(u, v);
+				std::tie(valid, dIdx) = images_dx[i]->FloatValueAt(u, v);
+				std::tie(valid, dIdy) = images_dy[i]->FloatValueAt(u, v);
                 if (gray == -1.0)
                     continue;
-                double dIdx = *PointerAt<float>(*images_dx[i], u, v);
-                double dIdy = *PointerAt<float>(*images_dy[i], u, v);
                 double invz = 1. / G(2);
                 double v0 = dIdx * fx * invz;
                 double v1 = dIdy * fy * invz;
@@ -676,11 +682,6 @@ std::vector<Image> MakeDepthMasks(
         auto mask_dilated = DilateImage(*mask,
                 option.half_dilation_kernel_size_for_discontinuity_map_);
         images_mask.push_back(*mask_dilated);
-
-        char filename[255];
-        sprintf(filename, "image_%03d.png", i);
-        WriteImage(filename, *mask_dilated);
-
     }
     return std::move(images_mask);
 }

--- a/src/Core/ColorMap/ColorMapOptimization.cpp
+++ b/src/Core/ColorMap/ColorMapOptimization.cpp
@@ -175,14 +175,14 @@ std::tuple<bool, T> QueryImageIntensity(
     float u, v, depth;
     std::tie(u, v, depth) = Project3DPointAndGetUVDepth(V, camera, camid);
     if (img.TestImageBoundary(u, v, IMAGE_BOUNDARY_MARGIN)) {
-		int u_round = int(round(u));
-		int v_round = int(round(v));
+        int u_round = int(round(u));
+        int v_round = int(round(v));
         if (ch == -1) {
             return std::make_tuple(true,
-					*PointerAt<T>(img, u_round, v_round));
+                    *PointerAt<T>(img, u_round, v_round));
         } else {
             return std::make_tuple(true,
-					*PointerAt<T>(img, u_round, v_round, ch));
+                    *PointerAt<T>(img, u_round, v_round, ch));
         }
     } else {
         return std::make_tuple(false, 0);
@@ -201,8 +201,8 @@ std::tuple<bool, T> QueryImageIntensity(
         Eigen::Vector2d uv_shift = field.GetImageWarpingField(u, v);
         if (img.TestImageBoundary(uv_shift(0), uv_shift(1),
                 IMAGE_BOUNDARY_MARGIN)) {
-			int u_shift = int(round(uv_shift(0)));
-			int v_shift = int(round(uv_shift(1)));
+            int u_shift = int(round(uv_shift(0)));
+            int v_shift = int(round(uv_shift(1)));
             if (ch == -1) {
                 return std::make_tuple(true,
                         *PointerAt<T>(img, u_shift, v_shift));
@@ -349,10 +349,10 @@ void OptimizeImageCoorNonrigid(
                 if (!images_gray[i]->TestImageBoundary(uu, vv,
                         IMAGE_BOUNDARY_MARGIN))
                     continue;
-				bool valid; double gray, dIdfx, dIdfy;
-				std::tie(valid, gray) = images_gray[i]->FloatValueAt(uu, vv);
-				std::tie(valid, dIdfx) = images_dx[i]->FloatValueAt(uu, vv);
-				std::tie(valid, dIdfy) = images_dy[i]->FloatValueAt(uu, vv);
+                bool valid; double gray, dIdfx, dIdfy;
+                std::tie(valid, gray) = images_gray[i]->FloatValueAt(uu, vv);
+                std::tie(valid, dIdfx) = images_dx[i]->FloatValueAt(uu, vv);
+                std::tie(valid, dIdfy) = images_dy[i]->FloatValueAt(uu, vv);
                 Eigen::Vector2d dIdf(dIdfx, dIdfy);
                 Eigen::Vector2d dfdx = ((grids[2] - grids[0]) * (1 - q) +
                         (grids[3] - grids[1]) * q) / anchor_step;
@@ -495,10 +495,10 @@ void OptimizeImageCoorRigid(
                 if (!images_gray[i]->TestImageBoundary(u, v,
                         IMAGE_BOUNDARY_MARGIN))
                     continue;
-				bool valid; double gray, dIdx, dIdy;
-				std::tie(valid, gray) = images_gray[i]->FloatValueAt(u, v);
-				std::tie(valid, dIdx) = images_dx[i]->FloatValueAt(u, v);
-				std::tie(valid, dIdy) = images_dy[i]->FloatValueAt(u, v);
+                bool valid; double gray, dIdx, dIdy;
+                std::tie(valid, gray) = images_gray[i]->FloatValueAt(u, v);
+                std::tie(valid, dIdx) = images_dx[i]->FloatValueAt(u, v);
+                std::tie(valid, dIdy) = images_dy[i]->FloatValueAt(u, v);
                 if (gray == -1.0)
                     continue;
                 double invz = 1. / G(2);

--- a/src/Core/ColorMap/ColorMapOptimization.cpp
+++ b/src/Core/ColorMap/ColorMapOptimization.cpp
@@ -178,10 +178,10 @@ std::tuple<bool, T> QueryImageIntensity(
 		int u_round = int(round(u));
 		int v_round = int(round(v));
         if (ch == -1) {
-            return std::make_tuple(true, 
+            return std::make_tuple(true,
 					*PointerAt<T>(img, u_round, v_round));
         } else {
-            return std::make_tuple(true, 
+            return std::make_tuple(true,
 					*PointerAt<T>(img, u_round, v_round, ch));
         }
     } else {
@@ -201,8 +201,8 @@ std::tuple<bool, T> QueryImageIntensity(
         Eigen::Vector2d uv_shift = field.GetImageWarpingField(u, v);
         if (img.TestImageBoundary(uv_shift(0), uv_shift(1),
                 IMAGE_BOUNDARY_MARGIN)) {
-			int u_shift = int(uv_shift(0));
-			int v_shift = int(uv_shift(1));
+			int u_shift = int(round(uv_shift(0)));
+			int v_shift = int(round(uv_shift(1)));
             if (ch == -1) {
                 return std::make_tuple(true,
                         *PointerAt<T>(img, u_shift, v_shift));
@@ -350,7 +350,7 @@ void OptimizeImageCoorNonrigid(
                         IMAGE_BOUNDARY_MARGIN))
                     continue;
 				bool valid; double gray, dIdfx, dIdfy;
-				std::tie(valid, gray) = images_gray[i]->FloatValueAt(uu, vv); 
+				std::tie(valid, gray) = images_gray[i]->FloatValueAt(uu, vv);
 				std::tie(valid, dIdfx) = images_dx[i]->FloatValueAt(uu, vv);
 				std::tie(valid, dIdfy) = images_dy[i]->FloatValueAt(uu, vv);
                 Eigen::Vector2d dIdf(dIdfx, dIdfy);

--- a/src/Core/ColorMap/ColorMapOptimization.h
+++ b/src/Core/ColorMap/ColorMapOptimization.h
@@ -42,7 +42,7 @@ public:
             bool non_rigid_camera_coordinate = false,
             int number_of_vertical_anchors = 16,
             double non_rigid_anchor_point_weight = 0.316,
-            double maximum_iteration = 500,
+            double maximum_iteration = 300,
             double maximum_allowable_depth = 2.5,
             double depth_threshold_for_visiblity_check = 0.03,
             double depth_threshold_for_discontinuity_check = 0.1,

--- a/src/Python/Tutorial/Advanced/color_map_optimization.py
+++ b/src/Python/Tutorial/Advanced/color_map_optimization.py
@@ -50,7 +50,7 @@ if __name__ == "__main__":
     # Q.-Y. Zhou and V. Koltun,
     # Color Map Optimization for 3D Reconstruction with Consumer Depth Cameras,
     # SIGGRAPH 2014
-    option.maximum_iteration = 500
+    option.maximum_iteration = 300
     option.non_rigid_camera_coordinate = True
     color_map_optimization(mesh, rgbd_images, camera, option)
     draw_geometries([mesh])

